### PR TITLE
Adding host headers to the GQL calls from buildpack.

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Adding host headers to the GQL calls from buildpack to resolve those requests.

There is another CICD PR we need to merge along with this: https://github.com/magento-commerce/pwa-studio-cicd/pull/78

## Related Issue

Closes PWA-1987

## Acceptance

Scaffolding job should not fail in the CI.

### Verification Stakeholders

@sharkySharks 

## Verification Steps

1. Check the job once this is merged for the job to consider using new code.
Meanwhile, if you need to test this code, re-run this [new temporary job that is a replica of the original scaffolding job](https://console.aws.amazon.com/codesuite/codebuild/913454440186/projects/pwa-scaffold-build-test/history?region=us-east-1&builds-meta=eyJmIjp7InRleHQiOiIifSwicyI6e30sIm4iOjIwLCJpIjowfQ#).

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by sharkySharks** [comment]
LGTM - maybe update the pr title and the branch since it seems to be out of date wit the base. I don't see any `TODO` anywhere that the Danger check is barking about.

File: packages/pwa-buildpack/lib/Utilities/graphQL.js [hunk 1]
```diff
@@ -9,22 +11,35 @@ const fetchQuery = query => {
...
+        Accept: 'application/json',
+        'User-Agent': 'pwa-buildpack',
+        Host: targetURL.host
     };
 
     if (process.env.STORE_VIEW_CODE) {
         headers['store'] = process.env.STORE_VIEW_CODE;
     }
 
+    debug('Fetching query: %s', query);
```
Comment (sharkySharks) [inline packages/pwa-buildpack/lib/Utilities/graphQL.js:24]: i think let's remove the debug statements now, eh? and just keep the headers.

File: packages/pwa-buildpack/lib/Utilities/graphQL.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const debug = require('../util/debug').makeFileLogger(__filename);
```
Comment (sharkySharks) [inline packages/pwa-buildpack/lib/Utilities/graphQL.js:1]: do we want to keep all these debug statements in this file too?

File: packages/pwa-buildpack/lib/Utilities/graphQL.js (hunk 1 above)
Comment (sharkySharks) [reply packages/pwa-buildpack/lib/Utilities/graphQL.js:24]: this one?

File: packages/pwa-buildpack/lib/Utilities/graphQL.js [hunk 3]
```diff
@@ -9,22 +11,35 @@ const fetchQuery = query => {
...
+    debug('Fetching query: %s', query);
+
     return fetch(targetURL.toString(), {
         agent: targetURL.protocol === 'https:' ? httpsAgent : null,
         body: JSON.stringify({ query }),
         headers: headers,
         method: 'POST'
     })
-        .then(result => result.json())
+        .then(result => {
```
Comment (sharkySharks) [inline packages/pwa-buildpack/lib/Utilities/graphQL.js:32]: can probably just leave this as the above `.then(result => result.json())`

File: packages/pwa-buildpack/lib/Utilities/graphQL.js (hunk 1 above)
Comment (revanth0212) [reply packages/pwa-buildpack/lib/Utilities/graphQL.js:24]: These debug statements only run if the user whats them to. They are not console logs for instance. We usually have these debug lines in buildpack files to debug them when needed. For instance in this case I want to debug this file ill run the file with `DEBUG="*pwa-buildpack/lib/Utilities/graphQL*"`.

File: packages/pwa-buildpack/lib/Utilities/graphQL.js (hunk 2 above)
Comment (revanth0212) [reply packages/pwa-buildpack/lib/Utilities/graphQL.js:1]: It's a file logger object definition.

**Review by sharkySharks — APPROVED** [review]
lgtm

**Pull request merged**